db/storage.go

```diff
@@ -0,0 +1,91 @@
+package db
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/luciodesimone/golang-bootcamp/beer"
+	"io/ioutil"
+	"os"
+	"sync"
+)
+
+//RWMap contains the database structure and locks to prevent data
+//dependency on concurrent access
+type RWMap struct {
+	sync.RWMutex
+	m map[string]beer.Beer
+}
+
+type storage struct {
+	db         *RWMap
+	DB         *RWMap
+	connString string
+	connection *os.File
+}
+
+//New creates a storage with the
+func New(connString string, connection *os.File) DB {
+	return &storage{
+		db:         &RWMap{m: make(map[string]beer.Beer)},
+		connString: connString,
+		connection: connection,
```

Comment:
It's a good idea to accept a file.

It would be better if, in this exercise, we let the main func open the file and defer it's closing.

The DB would have to only use it, not control it.

Something like
```
    f, err := os.Open(filename)
    if err != nil {
        return "", err
    }
    defer f.Close()  // f.Close will run when we're finished.
```

On the main and give the f to the db